### PR TITLE
Msw edit activities

### PR DESCRIPTION
**Pull request by michaelwehrley**
Automatically Review Activities & Update Editability

* When a participant updates an activity from  
  the 'Your Activities' page, the activity is  
  automatically reviewed.
* Model test for ```update_as_reviewed```.
* Controller spec for success and errors.
* On the 'Your Activities' page, only activities  
  that have not been  reviewed and  
  incomplete can be edited.
* Update model spec.

[Finishes: #88824148]

### CODE REVIEW

File: spec/controllers/think_feel_do_engine/participants/activities_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      describe "PUT update" do
+        let(:activity) { double("activity") }
+        let(:participant) { double("participant") }
+
+        context "activity updates with update_as_reviewed" do
+          before do
+            sign_in_participant participant
+            allow(participant).to receive(:activities) { Activity }
+            allow(Activity).to receive(:find) { activity }
+            allow(activity).to receive(:update_as_reviewed) { true }
```
Comment (gwensmu) [inline spec/controllers/think_feel_do_engine/participants/activities_controller_spec.rb]: maybe allow this mock to receive the error chain as well, and then expect not to see errors

Comment (gwensmu) [inline spec/controllers/think_feel_do_engine/participants/activities_controller_spec.rb]: what are the conditions that determine if update_as_reviwed is true or false? how are they captured in this test?

File: spec/models/activity_spec.rb [hunk 2]
```diff
@@ -187,6 +187,24 @@ def sleeping(attributes = {})
...
   describe "instance methods" do
+    describe "#update_as_reviewed" do
+      let(:running) do
+        Activity.new(
+          participant: participants(:participant1),
+          activity_type: activity_types(:jogging)
+        )
+      end
+
+      it "updates activity with is_reviewed to be true" do
```
Comment (gwensmu) [inline spec/models/activity_spec.rb]: does anything need to happen to the activity for it to be marked is reviewed? eg, does the data change? or is it just the participant viewing the activity passively that counts as reviewing

File: spec/controllers/think_feel_do_engine/participants/activities_controller_spec.rb (hunk 1 above)
Comment (michaelwehrley) [inline spec/controllers/think_feel_do_engine/participants/activities_controller_spec.rb]: @gwensmu Does it work that I am checking that the notice is visible: #L27?

@gwensmu The conditions for update failing are the same as ```update``` in rails in general.  I did think about placing the model code: app/models/activity.rb#L123 straight in the controller, but I couldn't figure out a way to test it unless I checked the model object in the feature test which didn't seem right.  But I am open for other options.

File: spec/models/activity_spec.rb (hunk 2 above)
Comment (michaelwehrley) [inline spec/models/activity_spec.rb]: @gwensmu From my understanding, as long as they hit this route, that is, they are adding actual accomplishment and pleasure ratings.

Comment (gwensmu) [inline spec/models/activity_spec.rb]: @michaelwehrley cool makes sense

File: spec/controllers/think_feel_do_engine/participants/activities_controller_spec.rb (hunk 1 above)
Comment (gwensmu) [inline spec/controllers/think_feel_do_engine/participants/activities_controller_spec.rb]: @michaelwehrley it sounds like from below that reviewed should only be true if actual accomplishment and actual pleasure are not nil, but I also see a "reviewed and incomplete" condition -- I guess what I would like to see from this spec is a way to check that "update as reviewed" won't be successful unless the correct business logic is in place.

Comment (michaelwehrley) [inline spec/controllers/think_feel_do_engine/participants/activities_controller_spec.rb]: @gwensmu Good point. :tangerine:

**Comment by michaelwehrley** [comment]
@gwensmu are are these updates?  Thanks!

**Comment by gwensmu** [comment]
@michaelwehrley looks pretty complete to me! 🍢

**Pull request merged**